src/Parsers/ParserQuery.h

```diff
@@ -10,12 +10,21 @@ class ParserQuery : public IParserBase
 {
 private:
     const char * end;
+    ParserSettings parser_settings;
 
     const char * getName() const override { return "Query"; }
     bool parseImpl(Pos & pos, ASTPtr & node, Expected & expected) override;
 
 public:
-    explicit ParserQuery(const char * end_) : end(end_) {}
+    /// For tests/examples.
+    explicit ParserQuery(const char * end_)
+        : end(end_)
+    {}
+    /// For production parser.
+    explicit ParserQuery(const char * end_, const ParserSettings & parser_settings_)
+        : end(end_)
+        , parser_settings(parser_settings_)
+    {}
```

Comment:
Looks very strange, why do you need to change so common classes?
I thought you were going to just check your new setting `parse_settings_after_format` in `ParserInsertQuery` and that's all

Reply:
But how would you pass this setting to `ParserInsertQuery`?
Entrypoint for the parsing is - `ParserQuery`

Reply:
Probably it's better not to use the structure and just pass the parameter to constructors?

Reply:
Probably it's better not to use the structure and just pass the parameter to constructors?

Reply:
I would prefer to keep structure, it is always easier to extend structure then arguments.
But if you have strong objections, then OK, I can replace it with the variable.